Make Maven ITs less fragile

gsmet opened the pull request:
Enforcing versions of the plugins make sure we don't start downloading
them when running the tests which can lead to timeouts.
Also it makes the build more stable and actually tests the version we
are using when creating new projects.

Created as draft as I'd like Alexey's feedback.

To reproduce:
- get this PR
- drop this commit as it fixes the issue: https://github.com/quarkusio/quarkus/commit/d7922d9dbc2109d0b164c5dc2842108f7d783bba
- `mvn clean install`
- `mvn clean install -f integration-tests/maven -Dtest=DevMojoIT#testThatSourceChangesAreDetectedOnPomChange`

My understanding is that if `compile` is in the command line, we don't execute it ourselves but... when we are restarting dev mode due to a POM change, we should run `compile` again.

gsmet committed:
Make Maven ITs less fragile

Enforcing versions of the plugins make sure we don't start downloading
them when running the tests which can lead to timeouts.
Also it makes the build more stable and actually tests the version we
are using when creating new projects.

gsmet committed:
Not pushing compile in the command actually fixes this test

When running mvn compile quarkus:dev, compile is not executed when
restarting dev mode after the pom change. Not specifying compile makes
the compile goal to be run.

gsmet committed:
Try to stabilize the tests on Windows

gsmet commented:
Unfortunately, the last commit is problematic as I used valid ascii characters. I'm not completely sure what's going on with this test as I saw it failing in some of our regular builds, even without this change.

I'll revert the ascii one and see how CI goes.

aloubyansky commented on integration-tests/maven/src/test/java/io/quarkus/maven/it/DevMojoIT.java:
```diff
@@ -474,7 +474,10 @@ public void testThatSourceChangesAreDetectedOnPomChange() throws Exception {
                 .atMost(1, TimeUnit.MINUTES)
-                .until(() -> devModeClient.getHttpResponse("/app/hello").contains("hello " + uuid));
+                .until(() -> {
+                    System.out.println(devModeClient.getHttpResponse("/app/hello"));
```
Did you mean to keep it a part of the commit?

gsmet committed:
Disable JarRunnerIT#testNonAsciiDir on Windows

With maven-compiler-plugin 2.11.0, this test is not working anymore
on Windows:
Error while storing the mojo status: Input length = 1

gsmet committed:
Enable errors in Maven RunningInvoker

We need to see the details of the errors when something is going wrong.

gsmet committed:
Disable DevMojoIT#testExternalReloadableArtifacts on Windows

Installing the lib jar a second time leads to an access denied exception
on Windows, probably because the jar is still used by the dev mode
instance.
